lib/vmpooler/api/v1.rb

```diff
@@ -372,8 +376,16 @@ def atomically_allocate_vms(payload)
 
       payload = JSON.parse(request.body.read)
 
-      if all_templates_valid?(payload)
-        result = atomically_allocate_vms(payload)
+      if payload
+        invalid = invalid_templates(payload)
+        if invalid.empty?
+          result = atomically_allocate_vms(payload)
+        else
+          invalid.each do |bad_template|
+            metrics.increment('checkout.invalid.' + bad_template)
+          end
+          status 404
+        end
       else
         status 404
```

Comment:
Should we increment an `invalid` counter? Since payload is nil, do we lump them into the same bucket, e.g. `metrics.increment('checkout.invalid.<unknown>')`?